src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs

```diff
@@ -385,35 +385,26 @@ protected override void Generate(DropSchemaOperation operation, IModel model, Mi
         /// <param name="model"> The target model which may be <see langword="null"/> if the operations exist without a model. </param>
         /// <param name="builder"> The command builder to use to build the commands. </param>
         protected override void Generate(
-            CreateSequenceOperation operation,
-            IModel model,
-            MigrationCommandListBuilder builder)
+            [NotNull] CreateSequenceOperation operation,
+            [CanBeNull] IModel model,
+            [NotNull] MigrationCommandListBuilder builder)
         {
             Check.NotNull(operation, nameof(operation));
-            Check.NotNull(builder, nameof(builder));
 
```

Comment:
We should throw a `NotSupportedException`, if sequences are not supported. Something similar to:
https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/ca2d02fa2ee38c13c2eb2a56bbc4a0298e89f039/src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs#L847-L849
Finally, a supporting test that ensures that an appropriate exception is thrown when an unsupported server version is used, needs to be added as well.

Reply:
Yes, but implementing this, automatically breaks tests. 
When test are executed, the annotation [SupportedServerVersionFact(ServerVersion.SequencesSupportKey)] works as expected, since in Config.json I put "10.5.4-MariaDB" but _options.ServerVersion from MySqlMigrationsSqlGenerator  class has the value of 8.0.21-mysql (default), maybe it was never initialized?

Reply:
There is a `Generate` method I added just recently, that allows to specify a `MySqlDbContextOptionsBuilder`, that allows you to configure the server version:

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/b31b2acd4225cb927600282dff7f97d4eacddb79/test/EFCore.MySql.FunctionalTests/MigrationSqlGeneratorMySqlTest.cs#L1012-L1024

Used e.g. here:

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/b31b2acd4225cb927600282dff7f97d4eacddb79/test/EFCore.MySql.FunctionalTests/MigrationSqlGeneratorMySqlTest.MySql.cs#L141-L169

I merge it tomorrow into the `5.0-wip` branch, so you can use it.

Reply:
The commented part should be removed.

The message should also be changed to something more explicit, that hints at the fact that this is not a general error, but has to do with the currently used server version:

```
$"Cannot create sequence '{operation.Name}' because sequences are not supported in server version {_options.ServerVersion}."
```

The same would apply to the exception in `RestartSequenceOperation`.

Reply:
This applies to the following lines:

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/9232252659c39ad4016e4ae8696da807c001e227/src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs#L400-L402

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/9232252659c39ad4016e4ae8696da807c001e227/src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs#L281-L283